dGame/dComponents/RacingControlComponent.cpp

```diff
@@ -99,7 +98,9 @@ void RacingControlComponent::LoadPlayerVehicle(Entity* player,
 
 	if (item == nullptr) {
 		Game::logger->Log("RacingControlComponent", "Failed to find item");
-
+		auto* playerInstance = dynamic_cast<Player*>(player);
+		m_LoadedPlayers--;
```

Comment:
You'll also want to remove the player from the m_LobbyPlayers vector, that or just do the SendToZone and let the update method take care of churning the correct variables.